### DISCUSSION TITLE
Improve contact shadow

### DISCUSSION THREAD

**Discussion by atirutw** [godotengine/godot-proposals#4441] [3D]
While the editor tooltip says it can look broken, I think it's a huge understatement as even small values (0.1 - 0.2) looks super broken in some cases(also depends on light type). Contact shadow for directional light is the worst.

Contact shadows are useful for more detailed shadows(small meshes, detailed character, etc.) but it can't be used for even the smallest of detail in such a state.

**Comment by Calinou** [comment]
See my comment here: https://github.com/godotengine/godot-proposals/issues/3643#issuecomment-988317471

Contact shadows are probably fixable by an experienced developer (but they will remain expensive when enabled). That said, contact shadows won't be readded in 4.0 as they're obsoleted by shadow normal offset bias/shadow pancaking (which could also be backported to `3.x`).

**Reply by atirutw** [reply]
> which could also be backported to `3.x`

If that's gonna take a good while then I'd like to at least see a temporary fix for contact shadows.
